pkg/test/integration/server.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+
+	// Allow auth to cloud providers
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	syncPeriod = "30s"
+	errCleanup = "failure in athodyd default cleanup"
+)
+
+// OperationFn is a function that uses a Kubernetes client to perform and
+// operation
+type OperationFn func(*envtest.Environment, client.Client) error
+
+// Config is a set of configuration values for Athodyd setup.
+type Config struct {
+	CRDDirectoryPaths []string
+	Builder           OperationFn
+	Cleaner           OperationFn
+	ManagerOptions    manager.Options
+}
+
+// NewBuilder returns a new no-op Builder
+func NewBuilder() OperationFn {
+	return func(*envtest.Environment, client.Client) error {
+		return nil
+	}
+}
+
+// NewCleaner returns a new Cleaner that deletes all installed CRDs from the API
+// server.
+func NewCleaner() OperationFn {
```

Comment:
https://github.com/crossplaneio/crossplane/pull/1032#discussion_r346576526

Cross-linking this existing comment from the design doc.